### DISCUSSION TITLE
Specify font for cluster

### DISCUSSION THREAD

**Discussion by hyperknot** [maplibre/maplibre-gl-js#4737] [General]
@edent has a question regarding specifying the fonts for clustering
https://github.com/hyperknot/openfreemap/issues/10#issue-2540523386

I see it's trying to reach ["Open Sans Regular", "Arial Unicode MS Regular"] which 404s.

My questions:
- Can it be configured or is it hard coded?
- If it's hard-coded, can you change it? The problematic part is especially Arial Unicode MS Regular, which is a proprietary Microsoft font.

I see it's hosted on GitHub in various places, like in the MapLibre [demotiles](https://github.com/maplibre/demotiles/tree/gh-pages/font/Open%20Sans%20Regular%2CArial%20Unicode%20MS%20Regular) repo, but I'd prefer only to include open source fonts in my repos.

**Comment by HarelM** [comment]
Not sure I remember why this was closed, but it was discussed here I believe: 
https://github.com/maplibre/maplibre-style-spec/issues/311

Nevertheless, this is a style definition that can be overridden as far as I know.

**Comment by hyperknot** [comment]
Do you know where it might come from? The cluster example has this:

```json
{
    "id": "cluster-count",
    "type": "symbol",
    "source": "earthquakes",
    "filter": ["has", "point_count"],
    "layout": {
        "text-field": "{point_count_abbreviated}",
        "text-font": ["DIN Offc Pro Medium", "Arial Unicode MS Bold"],
        "text-size": 12
    }
}
```

But nothing about Open Sans, so it must be from another layer.

**Reply by HarelM** [reply]
I'll need a better reproduction scenario to actually be able to answer this. I would assume there is a layer in the style (not the cluster) that is using this font and maybe breaking the map or something else, IDK... Hard to tell without a proper reproduction...

**Reply by edent** [reply]
Looking through the source of https://unpkg.com/maplibre-gl/dist/maplibre-gl.js, it appears there's a hard-coded requirement for those fonts.

If you follow the tutorial for adding clusters to a map - https://maplibre.org/maplibre-gl-js/docs/examples/cluster/ - you'll see this:

```js
        map.addLayer({
            id: 'cluster-count',
            type: 'symbol',
            source: 'earthquakes',
            filter: ['has', 'point_count'],
            layout: {
                'text-field': '{point_count_abbreviated}',
                'text-font': ['DIN Offc Pro Medium', 'Arial Unicode MS Bold'],
                'text-size': 12
            }
        });
```

If you remove `'text-font': ['DIN Offc Pro Medium', 'Arial Unicode MS Bold'],`  the browser will request the fonts listed in maplibre-gl.js

Does that make sense?

**Reply by HarelM** [reply]
Yes, there is an implicit default font defined, which is open sans, as can be read in the issue I linked to.

**Reply by edent** [reply]
But the *other* default font is the proprietary "Arial Unicode MS Regular". That cannot be freely distributed.

**Reply by HarelM** [reply]
I agree, but all fonts can be overwritten, or am I missing something?

**Reply by hyperknot** [reply]
I checked it and there was no bug or error. Simply the font had to be changed in the official example to Noto Sans and everything worked fine.

I'll open a ticket about the non-open source fonts.
